performance/plugins/extrinsic3rdPartyClaim.ts

```diff
@@ -0,0 +1,204 @@
+        Assets.DEFAULT_AMOUNT.muln(40e6),
+        10,
+      ),
+    );
+    await this.mintTokensToUsers(testParams.threads, testParams.nodes, [
+      MGA_ASSET_ID,
+      await getLiquidityAssetId(MGA_ASSET_ID, new BN(this.tokens[0])),
+      await getLiquidityAssetId(MGA_ASSET_ID, new BN(this.tokens[1])),
+    ]);
+    await this.activate3rdPArtyRewards(testParams.nodes, [
+      await getLiquidityAssetId(MGA_ASSET_ID, new BN(this.tokens[0])),
+      await getLiquidityAssetId(MGA_ASSET_ID, new BN(this.tokens[1])),
+    ]);
+    // await this.mintERC20TokensToUsers(this.tokens, this.mgaNodeandUsers);
+    console.info(`Setup Done!`);
+    tokens = this.tokens;
+    return true;
+  }
+  async act(testParams: TestParams): Promise<boolean> {
+    await super.act(testParams);
+    //    const generator = (
+    //      users: { nonce: BN; keyPair: KeyringPair }[],
+    //      thread: number,
+    //      offset: BN,
+    //    ) => {
+    //      return createAndSignNewSchedules(users, thread, offset);
+    //    };
+    await runTransactions(
+      this.mgaNodeandUsers,
+      testParams,
+      createAndSignClaimRewards,
+    );
+    console.info(`.... Done Sending Txs`);
+    return true;
+  }
+
+  private async activate3rdPArtyRewards(nodes: string[], assets: BN[]) {
+    const activations: Promise<MangataGenericEvent[]>[] = [];
+    for (let nodeNumber = 0; nodeNumber < nodes.length; nodeNumber++) {
+      const node = nodes[nodeNumber];
+      const mga = await getMangata(node);
+      const usersWithLiq = this.mgaNodeandUsers.get(nodeNumber)!.users;
+      for (let i = 0; i < usersWithLiq.length; i++) {
+        const api = await mga.api();
+        for (const assetId of assets) {
+          console.info(
+            usersWithLiq[i].keyPair.address +
+              " activating " +
+              assetId +
+              usersWithLiq[i].nonce.toString(),
+          );
+          activations.push(
+            signTx(
+              api,
+              await ProofOfStake.activateLiquidityFor3rdpartyRewards(
+                assetId,
+                BN_ONE,
+                MGA_ASSET_ID,
+              ),
+              usersWithLiq[i].keyPair,
+              { nonce: usersWithLiq[i].nonce },
+            ),
+          );
+          usersWithLiq[i].nonce = usersWithLiq[i].nonce.addn(1);
+        }
+      }
+    }
+    testLog.getLog().info("Waiting for activations...");
+    await Promise.all(activations);
+    testLog.getLog().info("¡¡ Tokens activated !!");
+
+    return true;
+  }
+}
+
+async function createAndSignClaimRewards(
+  mgaSdk: MangataInstance,
+  users: { nonce: BN; keyPair: KeyringPair }[],
+  threadId: number,
+  nonceOffset: BN = new BN(0),
+) {
+  await mgaSdk.api();
+  setupUsers();
+  await setupApi();
+  const srcUser = users[threadId % users.length];
+  const nonce = srcUser.nonce.add(nonceOffset);
+
+  const assets = [tokens[0], tokens[1]];
+  const claimAssetId = threadId % 2 === 0 ? assets[0] : assets[1];
```

Comment:
threadId and claimAssetId are numbers, and when we use %2 for number, we receive 0 only if number is 0, so we can change this line to  `const claimAssetId = threadId === 0 ? assets[0] : assets[1];`

Reply:
Mmmm  isn’t the module operator? Ie 2%2 =0 ?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Remainder